### PR TITLE
fixed Resource.__repr__ method to avoid AttributeError. 

### PR DESCRIPTION
**Pull request by bdilday**
repr now prints elements nested under Resource

### CODE REVIEW

**Comment by mwcraig** [comment]
Restarted doc build fixed by #4547

File: astropy/io/votable/tree.py [hunk 1]
```diff
@@ -3014,9 +3014,7 @@ def __init__(self, name=None, ID=None, utype=None, type='results',
 
     def __repr__(self):
         buff = io.StringIO()
-        XMLWriter(buff).element(
-            self._element_name,
-            attrib=w.object_attrs(self, self._attr_list))
+        Resource.to_xml(self, XMLWriter(buff))
```
Comment (evertrol) [inline astropy/io/votable/tree.py]: Why not `self.to_xml(XMLWriter(buff))`?

**Comment by evertrol** [comment]
(This is largely a repeat from my comment on the original [astroquery issue # 610](https://github.com/astropy/astroquery/issues/610) )

I'm not sure if printing the full XML is the best solution. This can result in a quite lengthy output.

I personally don't mind the default Python symbolic output as given in the documentation linked above. If a user likes to get the full output, it's simple a matter of calling to_xml() on each Resource individually.

Looking at the history of the `__repr__` method, it looks like things went wrong at [PR #3639](https://github.com/astropy/astropy/pull/3639).  
In fact, the current `__repr__` (as implemented by that PR) tries not only to access a non-existent _element_name, but also uses the local variable w, which doesn't exist (unless it's some badly named global variable).

The simplest way to reproduce the error is
```python
from astropy.io.votable.tree import Resource                                                  
str(Resource())
```

File: astropy/io/votable/tree.py (hunk 1 above)
Comment (bdilday) [inline astropy/io/votable/tree.py]: No particular reason, I was following the convention of the __repr__method of SimpleElement

**Comment by bdilday** [comment]
Yes, I agree that the output is lengthy. I don't have a strong opinion on what the __repr__ method should do, I'd rather leave that decision to the people that use and contribute to astropy more than I do.

**Comment by bsipocz** [comment]
@pplim - I'm following up the status of the original astroquery issue, and wondering whether this can be merged. I don't get the error from that issue, but I think only because something else got broken in the meantime so I don't get any results back.

**Comment by pllim** [comment]
Uh, I wasn't involved in this thread, but looks like @evertrol is against changing the current behavior. I don't have a preference on how things are printed out.

**Comment by keflavich** [comment]
This issue is still affecting astroquery (see https://github.com/astropy/astroquery/issues/661).  Either this change should be merged, or the behavior of the votable should be fixed to handle votables missing these private attributes.

**Comment by bsipocz** [comment]
I agree with @keflavich, one way or the other, it would be great to have this fixed.

**Comment by bsipocz** [comment]
While this is not merged, the astroquery issue this is intended to solve is now fixed when using astropy master.

**Comment by pllim** [comment]
So, should we close this?

**Comment by bsipocz** [comment]
I think yes, as I understand the motivation behind this was to fix the bug that showed up in astroquery.

**Pull request closed without merging**